src/borg/testsuite/repo_new.py

```diff
@@ -0,0 +1,257 @@
+import os
+
+import pytest
+
+from ..helpers import IntegrityError
+from ..repository import Repository, MAX_DATA_SIZE, TAG_DELETE, TAG_PUT2, TAG_PUT, TAG_COMMIT
+from ..repoobj import RepoObj
+from .hashindex import H
+
+
+UNSPECIFIED = object()  # for default values where we can't use None
+
+
+def fchunk(data, meta=b""):
+    # create a raw chunk that has valid RepoObj layout, but does not use encryption or compression.
+    meta_len = RepoObj.meta_len_hdr.pack(len(meta))
+    assert isinstance(data, bytes)
+    chunk = meta_len + meta + data
+    return chunk
+
+
+def pchunk(chunk):
+    # parse data and meta from a raw chunk made by fchunk
+    meta_len_size = RepoObj.meta_len_hdr.size
+    meta_len = chunk[:meta_len_size]
+    meta_len = RepoObj.meta_len_hdr.unpack(meta_len)[0]
+    meta = chunk[meta_len_size : meta_len_size + meta_len]
+    data = chunk[meta_len_size + meta_len :]
+    return data, meta
+
+
+def pdchunk(chunk):
+    # parse only data from a raw chunk made by fchunk
+    return pchunk(chunk)[0]
+
+
+class RepositoryTestCaseBase:
+    def reopen(self, repository, exclusive=UNSPECIFIED):
+        repository.__exit__(None, None, None)
+        repository.__enter__()
+        repository.exclusive = exclusive
+
+    def add_keys(self, repository):
+        repository.put(H(0), fchunk(b"foo"))
+        repository.put(H(1), fchunk(b"bar"))
+        repository.put(H(3), fchunk(b"bar"))
+        repository.commit(compact=False)
+        repository.put(H(1), fchunk(b"bar2"))
+        repository.put(H(2), fchunk(b"boo"))
+        repository.delete(H(3))
```

Comment:
i don't see a call of this method.

Reply:
It gets called by some of the later tests in this file